### PR TITLE
addon-technology formulation

### PR DESCRIPTION
**Pull request by gidden**
# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added
- [x] Description in RELEASE_NOTES.md Added

# Adding to RELEASE_NOTES.md

Please add a single line in the release notes similar to the following:

```
- (#XX)[http://link-to-pr.com] Added feature which does something
```

### CODE REVIEW

File: message_ix/core.py [hunk 1]
```diff
@@ -58,6 +58,37 @@ def _init_scenario(s, commit=False):
...
+                (s.init_set, {'args': ('addon',)}),
+                (s.init_set, {'args': ('type_addon',)}),
+                (s.init_set, {'args': ('cat_addon', ['type_addon', 'addon'])}),
+                (s.init_set, {
+                    'args': ('map_tec_addon', ['technology', 'type_addon'])
+                }),
+                (s.init_par, {
+                    'args': (
+                        'addon_conversion',
+                        ['node', 'technology', 'addon',
```
Comment (danielhuppmann) [inline message_ix/core.py]: this should be indexed on `type_addon`, not `addon` - please double-check with @OFR-IIASA to make sure this is the latest version of the code that actually worked for the many-to-many addon mapping.

**Comment by gidden** [comment]
cherry-picked and rebased, should be good to look at @danielhuppmann

File: message_ix/model/MESSAGE/model_core.gms [hunk 2]
```diff
@@ -759,6 +761,112 @@ RENEWABLES_CAPACITY_REQUIREMENT(node,inv_tec,commodity,year)$(
...
+* jointly with a parent technology (e.g., abatement option, SO2 scrubber, power plant cooling technology).
+*
+*   .. math::
+*      \sum_{\substack{t' \sim t^A, y^V \leq y}} addon\_conversion_{n,t',y^V,y,m,h} \cdot ACT_{n,t',y^V,y,m,h}
+*          \leq
+*      addon\_up_{n,t^a,y,m,h,t^A} \cdot \sum_{\substack{t, y^V \leq y}} ACT_{n,t,y^V,y,m,h}
+*
+***
+
+$ONTEXT
```
Comment (danielhuppmann) [inline message_ix/model/MESSAGE/model_core.gms]: please remove this commented-out section...

File: message_ix/model/MESSAGE/model_core.gms [hunk 3]
```diff
@@ -759,6 +761,112 @@ RENEWABLES_CAPACITY_REQUIREMENT(node,inv_tec,commodity,year)$(
...
+* Constraints for addon technologies
+* ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+*
+* Equation ADDON_ACTIVITY_UP
+* """"""""""""""""""""""""""
+* This constraint provides an upper bound on the activity of an addon technology that can only be operated
+* jointly with a parent technology (e.g., abatement option, SO2 scrubber, power plant cooling technology).
+*
+*   .. math::
+*      \sum_{\substack{t' \sim t^A, y^V \leq y}} addon\_conversion_{n,t',y^V,y,m,h} \cdot ACT_{n,t',y^V,y,m,h}
```
Comment (danielhuppmann) [inline message_ix/model/MESSAGE/model_core.gms]: can you please update that the `addon_conversion` factor is now on the right-hand-side of the equation?

File: message_ix/model/MESSAGE/model_core.gms [hunk 4]
```diff
@@ -759,6 +761,112 @@ RENEWABLES_CAPACITY_REQUIREMENT(node,inv_tec,commodity,year)$(
...
+* Equation ADDON_ACTIVITY_LO
+* """"""""""""""""""""""""""
+* This constraint provides a lower bound on the activity of an addon technology that has to be operated
+* jointly with a parent technology (e.g., power plant cooling technology). The parameter `addon_lo` allows to define
+* a minimum level of operation of addon technologies relative to the activity of the parent technology.
+* If `addon_minimum = 1`, this means that it is mandatory to operate the addon technology at the same level as the
+* parent technology (i.e., full mitigate).
+*
+*   .. math::
+*      \sum_{\substack{t' \sim t^A, y^V \leq y}} addon\_conversion_{n,t',y^V,y,m,h} \cdot ACT_{n,t',y^V,y,m,h}
```
Comment (danielhuppmann) [inline message_ix/model/MESSAGE/model_core.gms]: as above, please move `add_conversion` to the right-hand side

**Comment by gidden** [comment]
done!

**Comment by gidden** [comment]
close in favor of #91

**Pull request closed without merging**